crates/client/db/src/da_db.rs

```diff
@@ -1,33 +1,24 @@
 use std::sync::Arc;
 
-// Substrate
 use parity_scale_codec::{Decode, Encode};
 use sp_database::Database;
-// Starknet
 use starknet_api::block::BlockHash;
-use starknet_api::hash::StarkFelt;
 
 use crate::{DbError, DbHash};
 
-// The fact db stores DA facts that need to be written to L1
 pub struct DaDb {
     pub(crate) db: Arc<dyn Database<DbHash>>,
 }
 
-// TODO: purge old cairo job keys
 impl DaDb {
-    pub fn last_proved_block(&self) -> Result<BlockHash, DbError> {
-        match self.db.get(crate::columns::DA, crate::static_keys::LAST_PROVED_BLOCK) {
-            Some(raw) => {
-                let felt = StarkFelt::decode(&mut &raw[..])?;
-                Ok(BlockHash(felt))
-            }
-            None => Err(DbError::ValueNotInitialized(
-                crate::columns::DA,
-                // Safe coze `LAST_PROVED_BLOCK` is utf8
-                unsafe { std::str::from_utf8_unchecked(crate::static_keys::LAST_PROVED_BLOCK) }.to_string(),
-            )),
-        }
+    pub fn last_proved_block(&self) -> Result<Option<BlockHash>, DbError> {
```

Comment:
we can delete this actually, not used anymore. or we can create an issue if you want to do it in another PR :)

Reply:
We will start using it in the future, when we read messages from l1 again, won't we?

Reply:
Hmm, I don't think we need the last proved block to read a message from the L1